File: solutions/ocaml/rna-transcription/1/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]
let to_rna_nucleotide = function 
  | `G -> `C | `C -> `G | `T -> `A | `A -> `U
let to_rna = List.map to_rna_nucleotide

File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

func ToRNA(dna string) string {
	m := map[rune]rune{'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}
	rna := []rune(dna)
	for i, nucleotides := range dna {
		rna[i] = m[nucleotides]
	}
	return string(rna)
}